department-of-veterans-affairs/caseflow
# Handle scheduled in error disposition from case details

Resolves https://vajira.max.gov/browse/CASEFLOW-442
Link to design: https://www.figma.com/file/h5gRVZxR2vP5gNJy92LfbV/Hearing-Task-Flows-and-Routing?node-id=601%3A181
### Description
- Create a new component `HearingScheduledInErrorModal` to display modal 
- Add new disposition logic in `AssignHearingDispositionTask`
- Write tests

### Testing Plan
1. Login as `BVATWARNER`
2. go to http://localhost:3000/hearings/schedule and pick a veteran to schedule for
3. go to case details page and schedule the veteran for a hearing
4. > The Actions dropdown has a new option for scheduled hearings: "Remove hearing to correct a scheduling error"
When a user selects that option, the same modal appears as described above for the Daily Docket page, and gives the user the following options: `Reschedule immediately` and `Send to Schedule Veterans list`
5. > Ability to edit/add to the hearing Notes field (same notes field that appears on Hearing Details and Daily Docket)
 - go to hearing details page and add notes to the hearing
 - come back to case details page and open the modal - expect to see the notes you just saved on hearing
 - Ensure you can edit the notes
6. > If the user chooses to "Send to Schedule Veterans list" and clicks the "Submit" button, they remain on the Case Details page with a success alert at the top of the page
7. > After the new hearing disposition has been confirmed, the hearing remains in the Case Details > Hearings section and shows the new disposition
 - reload the page and see that the hearings section shows as expected
 - go to hearing details page and expect to see the new edited notes
8. > Removed from Daily Docket
- go to daily docket that the hearing was scheduled for and ensure that the hearing is no longer there
9. > Change shows up in Case Timeline similar to how it does today for postponed hearings
 - no changes to timeline
10. Choose to `Reschedule Immediately` on the same appeal
  - choose `video`
     - fill out the form (leave something blank and expect error messages)
     - hit `submit` 
     - repeat steps 7 to 9
   - choose `virtual`
     - fill out form (enter invalid emails and expect error messages)
     - hit `submit` 
     - repeat steps 7 to 9
  11. on the same appeal (in case details), click on action `Send for hearing disposition change` and confirm
  12. go to Hearing Admin queue http://localhost:3000/organizations/hearing-admin and choose the task you just created and go to case details page
  13. on case details, click on action `Change hearing disposition` and choose `Scheduled In Error` disposition and click confirm
  14. repeat steps 7 to 9
 
### User Facing Changes
<img width="540" alt="Screen Shot 2021-02-03 at 10 14 03 PM" src="https://user-images.githubusercontent.com/20735998/106916510-4369b380-66d5-11eb-8a2b-351d91fbf284.png">
<img width="524" alt="Screen Shot 2021-02-03 at 10 15 10 PM" src="https://user-images.githubusercontent.com/20735998/106839995-49c44500-666d-11eb-8c27-0920f0bb8fee.png">
<img width="543" alt="Screen Shot 2021-01-29 at 5 06 59 PM" src="https://user-images.githubusercontent.com/20735998/106840116-7ed09780-666d-11eb-9108-d7d27c9f82f3.png">
